lib/commands/serve.js

```diff
@@ -46,11 +46,11 @@ module.exports = Command.extend({
 
         this.ui.writeDeprecateLine(
           'Using the `baseURL` setting is deprecated, use `rootURL` instead.',
-          !(!config.rootURL && config.baseURL));
+          !((typeof config.rootURL !== 'string') && config.baseURL));
```

Comment:
These warnings should change to `in` checks (as we don't really care about the values, just that the property is present).